File: sdk/typescript/src/tools.ts
/** Tool definitions, embedded verbatim from ../../tools.json at build time (one definition, every surface). */
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";

export interface ToolDef { name: string; description: string; input_schema: Record<string, any> }

function load(): { version: string; tools: ToolDef[] } {
  const here = dirname(fileURLToPath(import.meta.url));
  for (const candidate of [join(here, "tools.json"), join(here, "..", "tools.json"), join(here, "..", "..", "..", "tools.json")]) {
    try { return JSON.parse(readFileSync(candidate, "utf8")); } catch { /* try next */ }
  }
  throw new Error("tools.json not found next to the package; run `npm run build`");
}

const loaded = load();
export const TOOLS_VERSION: string = loaded.version;
export const TOOLS: ToolDef[] = loaded.tools;

File: sdk/go/tools.go
package contextpull

import (
	_ "embed"
	"encoding/json"
)

//go:embed tools.json
var toolsJSON []byte

// ToolDef is one entry of tools.json: the shared tool contract for every surface.
type ToolDef struct {
	Name        string         `json:"name"`
	Description string         `json:"description"`
	InputSchema map[string]any `json:"input_schema"`
}

// Tools returns the tool definitions embedded from tools.json (kept in sync with the repo root by CI).
func Tools() (version string, tools []ToolDef) {
	var doc struct {
		Version string    `json:"version"`
		Tools   []ToolDef `json:"tools"`
	}
	_ = json.Unmarshal(toolsJSON, &doc)
	return doc.Version, doc.Tools
}
